applications/StructuralMechanicsApplication/tests/test_dynamic_schemes.py

```diff
@@ -45,25 +44,37 @@ def _base_spring_test_pseudo_static_scheme(self, current_model, scheme_name = "p
 
         current_disp = 0.0
         current_vel = init_velocity
+        c1 = 200.0
+        c4 = 1.0
         # Solve the problem
-        while(time <= end_time):
+        while time <= end_time:
             time = time + dt
             step = step + 1
             mp.CloneTimeStep(time)
             mp.ProcessInfo[KratosMultiphysics.STEP] = step
 
             self.strategy.Solve()
 
-            current_analytical_displacement_y = current_disp + current_vel * dt - current_vel * beta * mass * dt/2.0
-            current_disp = current_analytical_displacement_y
-            current_analytical_velocity_y = current_vel - current_vel * beta * mass
+            delta_disp = 0.0
+            if beta > 0.0 and current_vel > 0.0:
+                rhs = beta * mass * current_vel
+                lhs = beta * mass * c1
+                delta_disp = rhs/lhs
+                current_vel = c1 * delta_disp - c4 * current_vel
+
+            current_analytical_velocity_y = current_vel
             current_vel = current_analytical_velocity_y
+            current_analytical_displacement_y = current_disp + current_vel * dt + delta_disp
+            current_disp = current_analytical_displacement_y
+
+            disp = node.GetSolutionStepValue(KratosMultiphysics.DISPLACEMENT_Y,0)
+            vel = node.GetSolutionStepValue(KratosMultiphysics.VELOCITY_Y,0)
 
             ## ASSERT
-            self.assertAlmostEqual(node.GetSolutionStepValue(KratosMultiphysics.DISPLACEMENT_Y,0), current_analytical_displacement_y, delta=1e-3)
-            self.assertAlmostEqual(node.GetSolutionStepValue(KratosMultiphysics.VELOCITY_Y,0), current_analytical_velocity_y, delta=1e-3)
+            self.assertAlmostEqual(disp, current_analytical_displacement_y, delta=1e-12)
+            self.assertAlmostEqual(vel, current_analytical_velocity_y, delta=1e-12)
 
-    def _base_spring_test_dynamic_schemes(self, current_model, scheme_name = "bossak", buffer_size = 2, dt = 5.0e-3):
+    def _base_spring_test_dynamic_schemes(self, current_model, scheme_name = "bossak", buffer_size = 2, dt = 5.0e-3, tolerance = 1.0e-6):
```

Comment:
why are u using a larger tolerance here than in the other examples?

Reply:
Because some schemes are less accurate, Euler is first order, BDF2 is second order

Reply:
In this particular case I don't know, but 1.0e-6 is good precision

Reply:
you used 1e-9 elsewhere. Is it too much here?

Reply:
Yes

Reply:
ok